src/main.py

```diff
@@ -39,128 +56,170 @@ async def on_close() -> None:
         await bc.finish(say_goodbye=False)
 
 
-@client.event
-async def on_message(message: Message) -> None:
-    global controllers
-    global loaded_image
+# Allow only one async routine to calculate list at a time
+list_task_control_lock = asyncio.Lock()
+
 
-    if message.author == client.user:
+# List command
+@client.slash_command()
+@application_checks.has_role(config.dj_role_name)
+async def list(
+    interaction: Interaction,
+    list_channel: Optional[TextChannel] = SlashOption(required=False),
+) -> None:
+    """Download and list all media sent in a chosen text channel."""
+    bc = get_controller(interaction.guild_id)
+
+    if bc and bc.is_active():
+        await interaction.response.send_message("We're busy busting.", ephemeral=True)
         return
 
-    # Do not process messages outside of guild text channels
-    if not isinstance(message.channel, TextChannel):
+    # Give up if locked
+    if list_task_control_lock.locked():
+        await interaction.response.send_message(
+            "A list is already in progress.", ephemeral=True
+        )
         return
 
-    # The message author must be a guild member, so that we can
-    # check if they have the appropriate role below
-    if not isinstance(message.author, Member):
+    if list_channel is None:
+        list_channel = interaction.channel
+
+    async with list_task_control_lock:
+        await interaction.response.defer(ephemeral=True)
```

Comment:
I think this makes the bot say "thinking..." right? If so, it'd be good to label that with a comment since it isn't immediately obvious from the code.